meshnet/server.go

```diff
@@ -318,12 +319,11 @@ func (s *Server) DisableMeshnet(context.Context, *pb.Empty) (*pb.MeshnetResponse
 		s.pub.Publish(fmt.Errorf("disabling fileshare: %w", err))
 	}
 
-	// stop networker only if mesh peer connected before
-	if s.isPeerConnected {
+	// stop networker only if we are connected to mesh peer
+	if s.wasPeerConnected && s.netw.LastServerName() == s.lastConnectedPeer {
```

Comment:
2 more things:
1. I don't think `wasPeerConnected` is necessary (sorry if I hadn't made that clear before). Well maybe in the case when no connections were made at all, maybe then both would be equal to empty string, which brings me to the second point...
2. What happens if we connect to peer, disconnect and then disable mesh? It seems to me by this logic that it will still try to stop the networker even though it is already stopped. Nothing bad will happen it seems, only an error will be logged, but it would still be nice to avoid that.

Reply:
1. Yes, when no connections were made, both equal to empty strings, so I kept the boolean to avoid calling `Stop` in such case.
2. You are right, `Stop` in being called in such case. However, no error is emitted, because `Stop` checks with appropriate locks whether the VPN is connected before calling its actual implementation: https://github.com/NordSecurity/nordvpn-linux/blob/main/networker/networker.go#L412-L422

So maybe the `wasPeerConnected` should be removed and it should rely `if netw.isVpnSet` check inside `Stop` in both cases?

Reply:
At this point it's a very minor thing then, but I guess yeah, `wasPeerConnected` can be removed.